mri/reconstructors/self_calibrating.py

```diff
@@ -0,0 +1,180 @@
+# -*- coding: utf-8 -*-
+##########################################################################
+# pySAP - Copyright (C) CEA, 2017 - 2018
+# Distributed under the terms of the CeCILL-B license, as published by
+# the CEA-CNRS-INRIA. Refer to the LICENSE file or to
+# http://www.cecill.info/licences/Licence_CeCILL-B_V1-en.html
+# for details.
+##########################################################################
+
+"""
+This implements the self-calibrating reconstruction for the multi-channel case.
+"""
+
+# Module import
+from .base import ReconstructorBase
+from ..operators import GradSelfCalibrationSynthesis, \
+    GradSelfCalibrationAnalysis, WaveletN
+from .utils.extract_sensitivity_maps import get_Smaps
+
+
+class SelfCalibrationReconstructor(ReconstructorBase):
+    """ Self Calibrating reconstruction for multi-channel acquisition.
+    The coil sensitivity is estimated from a small portion of the  k-space
+    center and used to reconstruct the complex image.
+
+    For the Analysis case finds the solution for x of:
+        (1/2) * sum(||F Sl x - yl||^2_2, n_coils) + mu * ||Wt x||_1
```

Comment:
Please change the L1 norm at the end to make it more readable

Reply:
I am sorry, I didnt quite get you? You want me to make it as sum(Wt x) ?

Reply:
No I want to have the `H(Wt x)` the L1 norm here is very specific a,d it could be replaced with another norm like L2, the documentation should be more general than this

Reply:
Cool, makes sense